lib/atoi/strtoi/strton.h

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2023-2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_ATOI_STRTOI_STRTON_H_
+#define SHADOW_INCLUDE_LIB_ATOI_STRTOI_STRTON_H_
+
+
+#include <config.h>
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/param.h>
+
+#include "attr.h"
+
+
+#define template_STRTON(T)                                            \
+ATTR_STRING(1) ATTR_ACCESS(write_only, 2) ATTR_ACCESS(write_only, 6)  \
+inline T                                                       \
+STRTON(T,                                                             \
+    const char *s, char **restrict endp, int base,                    \
+    T min, T max, int *restrict status)                               \
+{                                                                     \
```

Comment:
Except for the attribute parameter references being off-by-one, this looks very cool.